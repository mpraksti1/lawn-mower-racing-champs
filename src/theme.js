const theme = {
  breakpoints: {
    mobile: 767,
    desktop: 768,
  },
  fonts: {
    baseFont: 'Helvetica',
    sansFont: 'Montserrat',
  },
  baseSizeUnit: 8,
  baseFontUnit: 16,
  colors: {
    yellow: '#f0ff0f',
    red: '#9d0404',
    gray: '#cfcfcf',
    darkGray: '#333',
    green: '#049d63',
    darkGreen: '#097149',
    white: '#eee',
    black: '#111',
    transparentBlack: 'rgba(0, 0, 0, 0.1)',
  },
  fontSizes: {
    xsm: '0.75rem',
    sm: '0.875rem',
    md: '1rem',
    lg: '1.125rem',
    xlg: '1.5rem',
    xxlg: '2.25rem',
  },
};

export default theme;
