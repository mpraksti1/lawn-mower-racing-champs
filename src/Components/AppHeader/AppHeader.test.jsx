import { AppHeader } from '.';
import theme from '../../theme';

const returnTrue = () => true;

describe('App header component', () => {
  it('renders without crashing', () => {
    shallow(<AppHeader
      history={fakeHistory}
      logoutUser={() => {}}
      checkAuthStatus={returnTrue}
    />);
  });

  it('matches snapshot', () => {
    const wrapper = shallow(
      <AppHeader
        theme={theme}
        history={fakeHistory}
        logoutUser={() => {}}
        checkAuthStatus={returnTrue}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('has the right logo text', () => {
    const wrapper = shallowWithTheme(
      <AppHeader
        theme={theme}
        history={fakeHistory}
        logoutUser={() => {}}
        checkAuthStatus={returnTrue}
      />, theme,
    );
    const logoText = wrapper.find('.logo-text').text();

    expect(logoText).toEqual('LMRC');
  });

  it('logs user out when button is clicked', () => {
    const spy = sinon.spy();
    const wrapper = shallowWithTheme(
      <AppHeader
        theme={theme}
        history={fakeHistory}
        logoutUser={spy}
        checkAuthStatus={returnTrue}
      />, theme,
    );

    wrapper.isLoggedIn = true;
    wrapper.hasValidToken = true;
    wrapper.find('.logout').simulate('click');

    expect(spy.calledOnce).toBe(true);
  });
});
