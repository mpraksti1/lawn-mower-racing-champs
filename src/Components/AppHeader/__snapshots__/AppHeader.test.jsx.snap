// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App header component matches snapshot 1`] = `
<AppHeaderStyled>
  <p
    className="logo"
  >
    <img
      alt="logo"
      className="logo-img"
      src="/assets/images/logo.png"
    />
    <span
      className="logo-text"
    >
      LMRC
    </span>
  </p>
  <nav
    className="nav"
  >
    <NavLink
      activeClassName="active"
      aria-current="page"
      className="nav-link"
      exact={true}
      to="/"
    >
      Home
    </NavLink>
    <NavLink
      activeClassName="active"
      aria-current="page"
      className="nav-link"
      to="/roster"
    >
      Roster
    </NavLink>
    <Button
      className="nav-link logout"
      inline={true}
      islink={true}
      onClick={[Function]}
    >
      Logout
    </Button>
  </nav>
</AppHeaderStyled>
`;
